### PR TITLE
How to build a Java GUI calculator from scratch using Box layout

### LINKED ISSUE #5916: How to build a Java GUI calculator from scratch using Box layout
### Writing sample(s): 
- [https://dev.to/princeibs](https://dev.to/princeibs)
- [https://www.section.io/engineering-education/working-with-gridbag-layout-in-java/](https://www.section.io/engineering-education/working-with-gridbag-layout-in-java/)

#### Proposed title of article
How to build a Java GUI calculator from scratch using Box layout

#### Proposed article introduction
The concept of building command line calculators is one of those projects every programmer loves to engage with. This is because it can be used to solve real life problems and computations. But limitations tend to come up when you want the calculator to be used by a user who is afraid of the black and white console. This is where GUIs come to play.
In this article, we will be building a calculator from scratch using the box layout to properly layout the components of the calculator. This calculator will be made of two parts, the GUI and the logic part - the part that performs computations. We will also get to learn how real life calculators work under the hood by incorporating functionalities like addition, subtraction, division, multiplication and many more. The article will also expose the readers to the power of box layout and how they can use it in other programs to convert their console apps to a GUI app.
  
#### Key takeaways 
By the end of this article, the readers will
* Get to know the logic that runs real life calculators
* Apply the logic in building a calculator in Java that can perform computations
* Gain a deep knowledge of a box layout and its functionalities
* Learn how to use the box layout to convert console apps to GUI

#### Article quality
Some days ago, I searched Google for assistance on [how to build a GUI calculator with Java using box layout](https://www.google.com/search?q=build+a+gui+calculator+with+java+using+box+layout&rlz=1C1CHBD_enNG902NG902&oq=build&aqs=chrome.0.69i59l3j69i57j69i60j69i61j69i65l2.5207j0j7&sourceid=chrome&ie=UTF-8) which is a course assignment for me. But I couldn't find any answer on the internet. This made me do more offline research on books, consultations, and some snippets of information I was able to get online. Because of the difficulty I encountered before solving the problem, I decided to write about it so as to help other fellow students who are in the same situation but couldn't find answers on the internet.

This article will cover building a fully fledged GUI calculator in Java using the box layout manager. It covers two main aspects which are - building the logic that runs the calculator, and building the calculator GUI using the box layout. 
There is currently no tutorial online that covers building a Java GUI calculator using the box layout. This article will go a long way to help college students who need a solution for assignment, and Java programmers who want to learn about box layout.
Below is a final look of the project - 
image

### PR DESCRIPTION
**Pull request by princeibs**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5916

### CODE REVIEW

**Comment by srishilesh** [comment]
Hey @princeibs ,

Thanks for submitting this PR.

Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you to get started with the review.

Meanwhile, we suggest you to go through approved PRs/published articles to see if you have missed out anything. This would help us avoid any potential closure of the PR.

Thanks.

**Comment by LinusMuema** [comment]
plagiarism check done ✅

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]
Hey @princeibs 👋.

My name is Briana, I'll be reviewing your article for publishing. I will make this review as fast and as swift as possible, with your help of course 😄 .

Good job on the article. It is a very interesting read. 

Just a few changes I need you to make as we continue with the preliminary revisions:
- Your article has quite a lot of grammatical mistakes and is quite unclear. I want you to use [Grammarly](https://app.grammarly.com/) to read through your article again and make these corrections.
- Label the steps as `Step one`  instead of `Part one`.
In case of any question or help needed please do not hesitate to ping me here or on Slack.
Also, notify me when you are done both here and on Slack.

Regards.

File: content/articles/how-to-build-a-java-gui-calculator-from-scratch-using-box-layout/index.md [hunk 1]
```diff
@@ -0,0 +1,733 @@
...
+
+
+* Check [here](https://github.com/princeibs/java-box-calculator/blob/97283e994097ba74ee67827b829bbf124d05b1e6/CalculatorBox.java) for the calculator source code at this point. 
+
+
+### Part 2: Building The Calculator Logic
+Now that we’ve completely built the user interface of the calculator. The next steps are to add functionalities to the buttons and also build the logic the calculator will run on.
+
+
+#### Creating the structure
```
Comment (BrianaNzivu) [inline content/articles/how-to-build-a-java-gui-calculator-from-scratch-using-box-layout/index.md:341]: Maybe you can number these steps? i.e
1. Creating the structure
2. .....

**Comment by princeibs** [comment]
Hi @BrianaNzivu 
Thank you for the suggestions. I have made the requested changes to the article.
Please let me know if the is still any more update to be made.

Thanks.

**Review by BrianaNzivu — APPROVED** [review]
Hey @princeibs  👋 .

Very good job on the article. It is a very interesting read 👏 .

Just a few things I'd like to point out:
-  It is always advisable to submit an article that is well polished. By this, I mean having an article that does not have too many grammatical corrections and functional code. I suggest that you check out our [resources](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) page which has links to various resourceful tools for writing. 
- Lastly please go through ALL the corrections I have made so that you can not repeat the same mistakes.

That's it! :smile: 

A final review will be done now by a content moderator.
Otherwise, congratulations on publishing your article🍾.

In case of any question or clarification (currently or in the future), do not hesitate to reach out via [Slack](https://app.slack.com/client/T5X14SH7G/D01A6MT673R) or my personal email (briananzivu@gmail.com). 
I am **ALWAYS** here to help.

It has been a pleasure reading and reviewing this article.

Looking forward to seeing you contribute more to EngEd. 🚀

Kind Regards.

**Comment by princeibs** [comment]
Thanks for the review @BrianaNzivu . I really learnt a lot from your corrections. 
I will definitely stick to the corrections in my future articles with Section.

CC: 👀 @hectorkambow @ahmadmardeni1

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
@princeibs 

Please reach out to me via Slack DM 👍  

Thank you

**Pull request merged**